### PR TITLE
Claims screen

### PR DESCRIPTION
**Pull request by FredrikAreschoug**
:baggage_claim:

### CODE REVIEW

File: owldroid/src/main/java/com/hedvig/android/owldroid/data/claims/ClaimsRepository.kt [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+class ClaimsRepository @Inject constructor(private val apolloClient: ApolloClient) {
+    private lateinit var claimsQuery: CommonClaimQuery
+
+    fun fetchCommonClaims(): Observable<CommonClaimQuery.Data?> {
+        claimsQuery = CommonClaimQuery
+            .builder()
+            .build()
+
+        return Rx2Apollo
+            .from(apolloClient.query(claimsQuery).watcher())
```
Comment (hedvigoscar) [inline owldroid/src/main/java/com/hedvig/android/owldroid/data/claims/ClaimsRepository.kt]:
```suggestion
            .from(apolloClient.query(claimsQuery))
```

File: owldroid/src/main/java/com/hedvig/android/owldroid/util/svg/SvgExtensions.kt [hunk 2]
```diff
@@ -0,0 +1,8 @@
+package com.hedvig.android.owldroid.util.svg
+
+import android.graphics.drawable.PictureDrawable
+import android.support.v4.app.Fragment
+
+fun Fragment.buildRequestBuilder() = GlideApp.with(this)
```
Comment (hedvigoscar) [inline owldroid/src/main/java/com/hedvig/android/owldroid/util/svg/SvgExtensions.kt]: We could perhaps inject this thing as the `.with(this)` refers to a context, and injected objects have access to `applicationContext`

File: owldroid/src/main/java/com/hedvig/android/owldroid/util/extensions/FragmentExtensions.kt [hunk 3]
```diff
@@ -20,14 +20,36 @@ fun Fragment.setupLargeTitle(
...
+    backgroundColor?.let {color ->
+        toolbar.setBackgroundColor(color)
+        collapsingToolbar.setBackgroundColor(color)
+    }
+
     icon?.let { toolbar.setNavigationIcon(it) }
     backAction?.let { toolbar.setNavigationOnClickListener { it() } }
 }
+
+fun Fragment.makeACall(uri: Uri) {
```
Comment (hedvigoscar) [inline owldroid/src/main/java/com/hedvig/android/owldroid/util/extensions/FragmentExtensions.kt]: ☎️

File: owldroid/src/main/java/com/hedvig/android/owldroid/ui/claims/ClaimsViewModel.kt [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+    val selectedSubViewData: MutableLiveData<CommonClaimQuery.CommonClaim> = MutableLiveData()
+
+    private val disposables = CompositeDisposable()
+
+    init {
+        fetchCommonClaims()
+    }
+
+    fun fetchCommonClaims() {
+        Timber.i("fetchCommonClaims OkHttp")
```
Comment (hedvigoscar) [inline owldroid/src/main/java/com/hedvig/android/owldroid/ui/claims/ClaimsViewModel.kt]:
```suggestion
```

File: owldroid/src/main/java/com/hedvig/android/owldroid/ui/claims/commonclaim/CommonClaimsAdapter.kt [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+import android.widget.ImageView
+import android.widget.TextView
+import com.bumptech.glide.RequestBuilder
+import com.hedvig.android.owldroid.R
+import com.hedvig.android.owldroid.graphql.CommonClaimQuery
+import kotlinx.android.synthetic.main.claims_common_claim_cell.view.*
+import org.jetbrains.annotations.NotNull
+import timber.log.Timber
+
+class CommonClaimsAdapter(private val commonClaims: @NotNull MutableList<CommonClaimQuery.CommonClaim>,
```
Comment (hedvigoscar) [inline owldroid/src/main/java/com/hedvig/android/owldroid/ui/claims/commonclaim/CommonClaimsAdapter.kt]: Use ktlint to format 👍

File: owldroid/src/main/java/com/hedvig/android/owldroid/ui/claims/commonclaim/EmergencyFragment.kt [hunk 6]
```diff
@@ -0,0 +1,48 @@
...
+        secondEmergencyButton.setOnClickListener {
+            makeACall(GLOBAL_ASSISTANCE_URI)
+        }
+        thirdEmergencyButton.setOnClickListener {
+            // todo: Write to hedvig
+        }
+    }
+
+    companion object {
+        private val GLOBAL_ASSISTANCE_URI = Uri.parse("tel:0123456789") // todo actual global assistance number
```
Comment (hedvigoscar) [inline owldroid/src/main/java/com/hedvig/android/owldroid/ui/claims/commonclaim/EmergencyFragment.kt]:
```suggestion
        private val GLOBAL_ASSISTANCE_URI = Uri.parse("tel:+4538489461") // todo actual global assistance number
```

File: owldroid/src/main/res/layout/fragment_claims.xml [hunk 7]
```diff
@@ -0,0 +1,87 @@
...
+            <TextView
+                style="@style/Headline.Large"
+                android:layout_width="match_parent"
+                android:layout_height="wrap_content"
+                android:layout_marginTop="@dimen/base_margin_triple"
+                android:gravity="center"
+                android:text="@string/CLAIMS_CREATE_CLAIM_TITLE" />
+
+            <TextView
+                style="@style/DialogParagraph"
```
Comment (hedvigoscar) [inline owldroid/src/main/res/layout/fragment_claims.xml]: 👮

File: owldroid/src/main/res/values/styles.xml [hunk 8]
```diff
@@ -87,13 +87,13 @@
         <item name="android:actionMenuTextAppearance">@style/AppTheme.MenuItem</item>
     </style>
 
-    <style name="DialogTitle">
+    <style name="SmallTitle">
```
Comment (hedvigoscar) [inline owldroid/src/main/res/values/styles.xml]: Are these represented in the text_styles, and if so can we replace them? 🙂

File: owldroid/src/main/java/com/hedvig/android/owldroid/util/svg/SvgExtensions.kt (hunk 2 above)
Comment (FredrikAreschoug) [reply owldroid/src/main/java/com/hedvig/android/owldroid/util/svg/SvgExtensions.kt]: Tried to fix this but it collides with the `RequestBuilder`, getting the error `e: error: cannot access RequestBuilder`. We can look at it together :)

File: owldroid/src/main/java/com/hedvig/android/owldroid/ui/claims/pledge/HonestyPledgeBottomSheet.kt [hunk 9]
```diff
@@ -17,7 +17,7 @@ class HonestyPledgeBottomSheet: RoundedBottomSheetDialogFragment() {
         dialog.setContentView(view)
 
         dialog.bottomSheetHonestyPledgeButton.setOnClickListener {
-            Toast.makeText(requireContext(), "Navigate to chat with the key'${arguments?.getString(ARGS_CLAIM_KEY)}'!", Toast.LENGTH_LONG).show()
+            Toast.makeText(requireContext(), "Navigate to chat with the key '${arguments?.getString(ARGS_CLAIM_KEY)}'!", Toast.LENGTH_LONG).show()
```
Comment (hedvigoscar) [inline owldroid/src/main/java/com/hedvig/android/owldroid/ui/claims/pledge/HonestyPledgeBottomSheet.kt]: 👷

File: owldroid/src/main/res/layout/claims_common_claim_cell.xml [hunk 10]
```diff
@@ -5,7 +5,7 @@
     android:layout_width="match_parent"
     android:layout_height="wrap_content"
     app:cardCornerRadius="8dp"
-    app:cardElevation="2dp"
+    app:cardElevation="8dp"
```
Comment (hedvigoscar) [inline owldroid/src/main/res/layout/claims_common_claim_cell.xml]: Should we move this to dimens? 🙂

**Pull request merged**